rp2040-hal/src/rosc.rs

```diff
@@ -65,6 +65,18 @@ impl RingOscillator<Disabled> {
             freq_hz: 6_500_000u32.Hz(),
         })
     }
+
+    /// Initializes the ROSC with a known frequency.
+    /// See sections 2.17.3. "Modifying the frequency", and 2.15.6.2. "Using the frequency counter"
+    /// in the rp2040 datasheet for guidance on how to do this before initialising the ROSC.
+    pub fn initialize_with_freq(self, known_freq_hz: u32) -> RingOscillator<Enabled> {
```

Comment:
The parameter could be a `HertzU32` instead of a `u32`:
```suggestion
    pub fn initialize_with_freq(self, known_freq: HertzU32) -> RingOscillator<Enabled> {
```

That way the caller does not have to check the docs or the parameter name to find out what the number means, but can instead pass something like `1_234_567.Hz()` or `1_234.kHz()`.

Reply:
Thank you both for the feedback!  I've now made changes to the function signature as requested, and added an example `rosc_as_system_clock`, which shows working code setting the ROSC to a desired target frequency.